pkg/lib/userexport/service.go

```diff
@@ -28,6 +29,117 @@ func NewLogger(lf *log.Factory) Logger {
+
+	address := mapGet[map[string]interface{}](user.StandardAttributes, "address")
+	record.Address = &Address{
+		Formatted:     mapGet[string](address, "formatted"),
+		StreetAddress: mapGet[string](address, "street_address"),
+		Locality:      mapGet[string](address, "locality"),
+		Region:        mapGet[string](address, "region"),
+		PostalCode:    mapGet[string](address, "postal_code"),
+		Country:       mapGet[string](address, "country"),
+	}
+
+	record.CustomAttributes = user.CustomAttributes
+
+	record.Roles = user.Roles
+	record.Groups = user.Groups
+
+	record.Disabled = user.IsDisabled
+
+	record.Identities = make([]*Identity, 0)
+	record.BiometricCount = 0
+	record.PasskeyCount = 0
+	for _, identity := range user.Identities {
+		switch identityType := identity.Type; identityType {
+		case model.IdentityTypeBiometric:
+			record.BiometricCount = record.BiometricCount + 1
+		case model.IdentityTypePasskey:
+			record.PasskeyCount = record.PasskeyCount + 1
+		case model.IdentityTypeLoginID:
+			record.Identities = append(record.Identities, &Identity{
+				Type: model.IdentityTypeLoginID,
+				LoginID: map[string]interface{}{
+					"key":            identity.LoginID.LoginIDKey,
+					"type":           identity.LoginID.LoginIDType,
+					"value":          identity.LoginID.LoginID,
+					"original_value": identity.LoginID.OriginalLoginID,
+				},
+				Claims: identity.LoginID.Claims,
+			})
+		case model.IdentityTypeLDAP:
+			record.Identities = append(record.Identities, &Identity{
+				Type: model.IdentityTypeLDAP,
+				LoginID: map[string]interface{}{
+					"server_name":             identity.LDAP.ServerName,
+					"last_login_username":     identity.LDAP.LastLoginUserName,
+					"user_id_attribute_name":  identity.LDAP.UserIDAttributeName,
+					"user_id_attribute_value": identity.LDAP.UserIDAttributeValue,
+					"attributes":              identity.LDAP.RawEntryJSON,
+				},
+				Claims: identity.LDAP.Claims,
+			})
+		case model.IdentityTypeOAuth:
+			record.Identities = append(record.Identities, &Identity{
+				Type: model.IdentityTypeOAuth,
+				LoginID: map[string]interface{}{
+					"provider_alias":      identity.OAuth.ProviderAlias,
+					"provider_type":       identity.OAuth.ProviderID.Type,
+					"provider_subject_id": identity.OAuth.ProviderSubjectID,
+					"user_profile":        identity.OAuth.UserProfile,
+				},
+				Claims: identity.OAuth.Claims,
+			})
+		}
+	}
+
+	record.Mfa = &MFA{
+		Emails:       make([]string, 0),
+		PhoneNumbers: make([]string, 0),
+		TOTPs:        make([]*MFATOTP, 0),
+	}
+	for _, authenticator := range user.Authenticators {
+		switch authenticatorType := authenticator.Type; authenticatorType {
+		case model.AuthenticatorTypeOOBEmail:
+			record.Mfa.Emails = append(record.Mfa.Emails, authenticator.OOBOTP.Email)
+		case model.AuthenticatorTypeTOTP:
+			record.Mfa.TOTPs = append(record.Mfa.TOTPs, &MFATOTP{
+				Secret: authenticator.TOTP.Secret,
+				URI:    authenticator.TOTP.UserID,
```

Comment:
This is incorrect. If you search for `GetURI`, you will see an example of how to generate a TOTP URI in pkg/lib/authenticationflow/declarative/intent_create_authenticator_totp.go

It takes an URIOptions. For issuer, you inject HTTPOrigin and use it as issuer. For accountName, you use User.EndUserAccountID.

Reply:
fixed the get uri logic